portfolio/src/main/webapp/script.js

```diff
@@ -66,71 +73,106 @@ function initTic() {
+          boxGrid[0][2].checked === boxGrid[1][1].checked &&
+          boxGrid[0][2].checked === boxGrid[2][0].checked) {
         return true;
       }
       return false;
     }
 
+    /** Displays winner message and removes the grid
+     *@param {string} symbol The symbol of the winner
+     */
     function gameOver(symbol) {
       paper.remove();
-      winnerMessage = document.createElement("p").appendChild(document.createTextNode(`The ${symbol} player wins`));
+      winnerMessage = document.createElement('p')
+          .appendChild(document.createTextNode(`The ${symbol} player wins`));
       tic.appendChild(winnerMessage);
     }
   }
-  
-  // removes the winner message and the grid to setup another turn, 
-  // reset button has another eventlistener calling the initTic function exactly after calling reset
+
+  /** removes the winner message and the grid to setup another turn,
+   * reset button has another eventlistener calling
+   * the initTic function exactly after calling reset
+   */
   function reset() {
-    if(winnerMessage !== null) 
+    if (winnerMessage !== null) {
       winnerMessage.remove();
+    }
     paper.remove();
   }
 }
 
 // call function when all elements of the window are loaded
-window.addEventListener('load',getMessageFromServer);
+window.addEventListener('load', getMessageFromServer);
 
-// adds message from server to html DOM using fetch()
+/** adds message from server to html DOM using fetch()*/
 function getMessageFromServer() {
-  let numberOfComments = document.getElementById('number-of-comments');
-  fetch('/data?numberOfComments='+numberOfComments.value).then(response => response.json()).then(messagesArray => {
-    let messageContainer = document.getElementById('message-container');
-    messageContainer.innerHTML='';
-    messagesArray.forEach(function(message) {
-      let listItem = document.createElement('li');
-      listItem.innerText = message;
-      messageContainer.appendChild(listItem);
-    });
-  });
+  const numberOfComments = document.getElementById('number-of-comments');
+  fetch('/data?numberOfComments='+numberOfComments.value)
+      .then((response) => response.json()).then((messagesArray) => {
+        const messageContainer = document.getElementById('message-container');
+        messageContainer.innerHTML = '';
+        messagesArray.forEach(function(message) {
+          const listItem = document.createElement('li');
+          listItem.innerText = message;
+          messageContainer.appendChild(listItem);
+        });
+      });
 }
-
+/** Clears all comments*/
 function clearComments() {
-  let request = new Request('\delete-data',{method: 'Post'});
+  const request = new Request('\delete-data', {method: 'Post'});
   fetch(request).then(getMessageFromServer);
 }
+/** Gets the login status*/
+function getLoginStatus() {
+  fetch('/login').then((response) => response.text())
+      .then((text) => {
+        return text === 'loggedIn';
```

Comment:
You're using the response here essentially as a boolean (either logged in or not), but the servlet is returning a string and it's not a good practice to rely on string comparison when other more structured alternatives are possible.
What about making the servlet return a JSON? You can have a field that is a boolean that says whether the user is logged in or not, and on top of that you can also add more fields in the future (e.g. email address).